src/clients/ntp-client_proposal.rb

```diff
@@ -289,16 +302,16 @@ def ui_init(rp, first_time)
 
     def AskUser
       ret = nil
-      ntp_server = Convert.to_string(UI.QueryWidget(Id(:ntp_address), :Value))
-      if !ValidateSingleServer(ntp_server)
-        ret = :invalid_hostname
-      else
+      if select_ntp_server
+        ntp_server = Convert.to_string(UI.QueryWidget(Id(:ntp_address), :Value))
+        return :invalid_hostname unless ValidateSingleServer(ntp_server)
+
         NtpClient.ntp_conf.clear_pools
         NtpClient.ntp_conf.add_pool(ntp_server)
-        retval = Convert.to_boolean(WFM.CallFunction("ntp-client"))
-        ret = :next if retval
-        MakeProposal()
       end
+      ret = :next if WFM.CallFunction("ntp-client")
```

Comment:
will it work when you just pick just one entry? Comment for method really missing.

Reply:
Yes, it is working. I have have added comments.